src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java

```diff
@@ -0,0 +1,138 @@
+                                zipFilesPathList.size());
+                        throw new IOException(errorMsg);
+                    }
+                    FilePath file = zipFilesPathList.get(0);
+                    context.logStatus("Deploy to app " + file.getBaseName() + " using file: " + file.getRemote());
+                    try (InputStream stream = file.read()) {
+                        String slotName = context.getSlotName();
+                        if (StringUtils.isEmpty(slotName)) {
+                            app.zipDeploy(stream);
+                        } else {
+                            DeploymentSlot slot = app.deploymentSlots().getByName(slotName);
+                            if (slot != null) {
+                                slot.zipDeploy(stream);
+                            } else {
+                                throw new IOException("Slot " + slotName + " not found");
+                            }
+                        }
+                    }
+                    break;
+                case WAR:
+                    aiDeployInfo = Constants.AI_WAR_DEPLOY;
+                    aiDeployFailedInfo = Constants.AI_WAR_DEPLOY_FAILED;
+                    for (FilePath filePath : files) {
+                        if (!filePath.getName().toLowerCase().endsWith(Constants.WAR_FILE_EXTENSION)) {
+                            context.logStatus(filePath.getName() + " is not WAR file. Will skip.");
+                            continue;
+                        }
+
+                        context.logStatus("Deploy to app " + filePath.getBaseName() + " using file: "
+                                + filePath.getRemote());
+
+                        try (InputStream stream = filePath.read()) {
+                            String slotName = context.getSlotName();
+                            if (StringUtils.isEmpty(slotName)) {
+                                app.warDeploy(stream, filePath.getBaseName());
+                            } else {
+                                DeploymentSlot slot = app.deploymentSlots().getByName(slotName);
+                                if (slot != null) {
+                                    slot.warDeploy(stream, filePath.getBaseName());
+                                } else {
+                                    throw new IOException("Slot " + slotName + " not found");
+                                }
+                            }
+                        }
+                    }
+                    break;
+                default:
+                    String errorMsg = String.format("Java app does not support Git deployment now. Please choose WAR"
+                            + " or ZIP deployment.", deployType.toString());
+                    throw new IOException(errorMsg);
+            }
+
+            context.setCommandState(CommandState.Success);
+            AzureAppServicePlugin.sendEvent(Constants.AI_WEB_APP, aiDeployInfo,
```

Comment:
is there way to save user's deploymentType? that will be useful telemetry data

Reply:
done.